packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java

```diff
@@ -623,10 +623,13 @@ public void maybeSetText(ReactTextUpdate reactTextUpdate) {
     if (reactTextUpdate.getText().length() == 0) {
       setText(null);
     } else {
-      // When we update text, we trigger onChangeText code that will
-      // try to update state if the wrapper is available. Temporarily disable
-      // to prevent an infinite loop.
+      boolean shouldKeepCompositeSpan = length() == spannableStringBuilder.length();
+
       getText().replace(0, length(), spannableStringBuilder);
+
+      if (shouldKeepCompositeSpan) {
```

Comment:
When I read through this at first, the name composite confused me, as meaning multiple spans together (e.g. nested spans or something else).

Could we consistently use composing instead?

Reply:
A second question, can there be multiple spans? It looks like we iterate through all of them, so maybe this should be named something like `shouldRestoreComposingSpans`.

Reply:
Yes, that makes a lot of sense, I should have used composing, I will adjust!

During my testing I haven't seen multiple composing spans, but it's added by auto complete, so I wouldn't take that as granted. So yes, I agree, the naming would make more sense. 

I will adjust and reply on this thread once that's done.

Reply:
@NickGerleman Adjusted naming as suggested, thank you!